bring/src/main/java/com/bobocode/petros/bring/factory/BeanFactory.java

```diff
@@ -0,0 +1,17 @@
+package com.bobocode.petros.bring.factory;
+
+import com.bobocode.petros.bring.context.domain.BeanDefinition;
+import com.bobocode.petros.bring.context.domain.BeanReference;
+
+import java.util.Map;
+
+public interface BeanFactory {
+
+    Map<String, BeanReference> getAllBeanReference();
+
+    BeanReference createBeanReference(String beanName);
```

Comment:
I think it will depend on for what purpose this map is planning to be used. Do we need any additional information that is stored in BeanDefinition? I think if we would need something additional we may add this info into BeanReference class since we add this wrapper for that reason as I understood

Reply:
I don't understand how we will create BeanReference using only beanName. We have to know beanType and scope at least.

Reply:
If we don`t need this methods in future I can remove it. About getting BeanReference by name we can do it I thought that we could get some BeanDefinition by accepted name and if we have more than one bean throw NoUniqueBeanDenitionExceoption.